src/app/components/fairChain/fairChain.component.ts

```diff
@@ -328,9 +375,77 @@ export class FairChainComponent implements OnInit {
   public redo(){
     this.updateData(this.undoRedoService.getSuccessorSnapshot())
   }
+
+  private showRelabelPopUp(pointer) {
+    this.nodeToRelableId = pointer.nodes[0];
+    let relabelRectOnDOM = this.getBoundingBoxOfNodeAsRectInDOM();
+    this.relabelPopUpInfo = this.cropAndTranslateRectInDOMToBeOverCanvasAndNode(relabelRectOnDOM);
+    this.nodeEdgeLabel = this.nodes.get(this.nodeToRelableId).label;
+
+    this.isShowingRelabelPopUp  = true;
+  }
+
+  private cropAndTranslateRectInDOMToBeOverCanvasAndNode(rect: RectOnDOM): RectOnDOM {
```

Comment:
Too long method name. To avoid this, you can add a description of the method which clearifies what it does